### PR TITLE
Adding initial ADR to template

### PR DESCRIPTION
**Pull request by jinder1s**
**Description:**

Describe in a couple of sentences what this PR adds

**JIRA:**

[XXX-XXXX](https://openedx.atlassian.net/browse/XXX-XXXX)

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Context
+-------
+
+TODO: Give context on what lead to the creation of this repo. What project is the repo related to?
+
+Decision
+--------
+
+TODO:
```
Comment (jinder1s) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:17]: I could not figure out what instructions to give here.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Consequences
+------------
+
+TODO:
```
Comment (jinder1s) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:22]: I could not figure out what instructions to give here.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Context
+-------
+
+TODO: Give context on what lead to the creation of this repo. What project is the repo related to?
+
+Decision
+--------
+
+TODO
```
Comment (jinder1s) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:17]: I could not figure out what instructions to give here.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Consequences
+------------
+
+TODO
```
Comment (jinder1s) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:22]: I could not figure out what instructions to give here.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst (hunk 3 above)
Comment (feanil) [reply python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:17]: `TODO: Clearly state how the context above leads you to creating this repo.`

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst (hunk 4 above)
Comment (feanil) [reply python-template/{{cookiecutter.placeholder_repo_name}}/docs/decisions/0001-purpose-of-this-repo.rst:22]: `TODO: As a result of this repo changing, what other things will change.`

**Pull request merged**